Unable to Update VS Code

Issue Type: <b>Bug</b>

When checking for updates, receive prompt saying the following:

"C:\Users\myusername\AppData\Local\Programs\Microsoft VS Code\_\chrome_100_percent.pak
An error occurred while trying to create a file in the destination directory: Access is denied"

Available options are:

- Try again
- Skip this file
- Cancel installation. 

First two options immediately bring prompt the error again, cancel installation suppresses it until auto-update re-attempts the installation.

Error occurs when running as administrator as well.

VS Code version: Code 1.66.1 (8dfae7a5cd50421d10cd99cb873990460525a898, 2022-04-06T14:50:12.141Z)
OS version: Windows_NT x64 10.0.19043
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 9 3900X 12-Core Processor             (24 x 3800)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.95GB (22.02GB free)|
|Process Argv|--crash-reporter-id 117dd477-59c8-4487-9ba4-aff765032dc6|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
prettier-vscode|esb|9.5.0


</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256860
pythonvspyl392:30443607
pythontb:30283811
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805:30301674
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30449410
pythonvsnew555:30457759
vscscmwlcmt:30465135
cppdebugcf:30466690

```

</details>

<!-- generated by issue reporter -->

...aaaand now my install is corrupt, great:

![image](https://user-images.githubusercontent.com/941331/163262542-827112e1-6193-4e5c-ac00-59bf1a11da46.png)

```
Uncaught Exception:
Error: Cannot find module 'C:\Users\alan\AppData\Local\Programs\Microsoft VS Code\resources\app\out\main'
    at Module._resolveFilename (node:internal/modules/cjs/loader:987:15)
    at Function.o._resolveFilename (node:electron/js2c/browser_init:253:840)
    at Object.<anonymous> (node:electron/js2c/browser_init:193:3263)
    at Object../lib/browser/init.ts (node:electron/js2c/browser_init:193:3507)
    at __webpack_require__ (node:electron/js2c/browser_init:1:128)
    at node:electron/js2c/browser_init:1:1200
    at node:electron/js2c/browser_init:1:1267
    at NativeModule.compileForInternalLoader (node:internal/bootstrap/loaders:312:7)
    at NativeModule.compileForPublicLoader (node:internal/bootstrap/loaders:252:10)
    at loadNativeModule (node:internal/modules/cjs/helpers:49:9)
```

--- a/build/win32/code.iss
+++ b/build/win32/code.iss
@@ -1494,7 +1494,7 @@ begin
   Permissions := '/grant:r "*S-1-5-18:(OI)(CI)F" /grant:r "*S-1-5-32-544:(OI)(CI)F" /grant:r "*S-1-5-11:(OI)(CI)RX" /grant:r "*S-1-5-32-545:(OI)(CI)RX"';
 
   #if "user" == InstallTarget
-    Permissions := Permissions + ' /grant:r "*S-1-3-0:(OI)(CI)F"';
+    Permissions := Permissions + Format(' /grant:r "*S-1-3-0:(OI)(CI)F" /grant:r "%s:(OI)(CI)F"', [GetUserNameString()]);
   #endif
 
   Exec(ExpandConstant('{sys}\icacls.exe'), ExpandConstant('"{app}" /inheritancelevel:r ') + Permissions, '', SW_HIDE, ewWaitUntilTerminated, ResultCode);

